lib/puma/cluster/worker.rb

```diff
@@ -122,49 +132,111 @@ def run
             debug_loaded_extensions "Loaded Extensions - worker 0:"
           end
 
-          stat_thread ||= Thread.new(@worker_write) do |io|
-            Puma.set_thread_name "stat pld"
-            base_payload = "#{PIPE_PING}#{Process.pid}"
-
-            while true
-              begin
-                b = server.backlog || 0
-                r = server.running || 0
-                t = server.pool_capacity || 0
-                m = server.max_threads || 0
-                rc = server.requests_count || 0
-                bt = server.busy_threads || 0
-                payload = %Q!#{base_payload}{ "backlog":#{b}, "running":#{r}, "pool_capacity":#{t}, "max_threads":#{m}, "requests_count":#{rc}, "busy_threads":#{bt} }\n!
-                io << payload
-              rescue IOError
-                Puma::Util.purge_interrupt_queue
-                break
-              end
-              sleep @options[:worker_check_interval]
-            end
-          end
+          make_sure_pinging(server)
+
           server_thread.join
         end
 
+        if @mold
```

Comment:
this is the new reforking logic, really the same as the `fork_worker` logic without the stop/restart complexity